internal/util/update/update_test.go

```diff
@@ -789,6 +790,210 @@ func TestCommand_Run_badStrategy(t *testing.T) {
 	}
 }
 
+func TestCommand_Run_subpackages(t *testing.T) {
+	testCases := []struct {
+		name            string
+		initialUpstream *dataset.Pkg
+		updatedUpstream *dataset.Pkg
+		updatedLocal    *dataset.Pkg
+		expectedLocal   *dataset.Pkg
+	}{
+		{
+			// TODO(mortent): This does not handle Kptfiles correctly.
```

Comment:
What do you mean by this comment? For this test case, `nestedbar` and `zork` should be fetched and Kptfiles and resources in foo and bar must be 3-way merged. You say that they are not being merged correctly?

Reply:
In this case the Kptfiles of the packages added in upstream are not added to the local fork.